### PR TITLE
[XGUARD-3542] task pane upgrade

### PR DESCRIPTION
**Pull request by SiamakSamie**
### Description
added tabview to taskpane with 5 tabs

- task placement and task info
- edit task: again this will look similar to to task modal edit tab in the kanban view. There is also a feature that when the user clicks cancel, it will revert all the changes they had done to the original version and trigger a message saying “Resetting Task Data“
- Task comments: We should be able edit/write/delete comments
- Related Task: We should be able to see related tasks as cards and be able to update/remove from group
- Task Logs: exactly like in kanban view tab

bugfix:
- fix issue with loading board/row/column in task placement
- fix dynamic update not working on kanban card when changing/removing group to card in kanban view

### What kind of change does this PR introduce?

- [x] New feature (adds functionality)
- [x] Bug fix (fixes an issue)
- [x] Refactor (code changes that do **NOT** change functionality)
- [ ] Code Coverage (adds Unit Tests)

### Checklist:
Please check if your PR fulfills the following requirements:

- [ ] Unit Tests for the changes have been added.
- [x] Contains **NO** breaking changes.
- [x] Associated with a JIRA issue.

### CODE REVIEW

File: src/resources/js/components/global/taskComponents/AddTaskData.vue [hunk 1]
```diff
@@ -293,6 +287,34 @@ export default {
...
+            if (this.cloneCardData.task_files.length)
+                this.checkedOptions.push('Upload Files');
+
+            this.getErpEmployees();
+            this.getContracts();
+            this.getBadges();
+            this.getMembers();
+            this.formatFilesForFilepond();
+        },
+
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/AddTaskData.vue:312]: Do you have spaces between all your functions or not, or variable

File: src/resources/js/components/global/taskComponents/AddTaskData.vue [hunk 2]
```diff
@@ -293,6 +287,34 @@ export default {
     },
     methods: {
 
+        getAndSetInitialData() {
+
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/AddTaskData.vue:291]: remove extra line

File: src/resources/js/components/global/taskComponents/AddTaskData.vue [hunk 3]
```diff
@@ -356,19 +377,32 @@ export default {
...
+                }
             }
         },
 
-        cancel(event) {
-            event.target.disabled = true;
-            this.eventHub.$emit("close-task-modal");
+        cancel() {
+
+            if (this.source === "KanbanTaskModal") {
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/AddTaskData.vue:398]: Consider an enum file for your kanban component names

File: src/resources/js/components/global/taskComponents/AddTaskData.vue [hunk 4]
```diff
@@ -356,19 +377,32 @@ export default {
...
+                    this.eventHub.$emit("update-task-card-data", this.cloneCardData);
+                }
             }
         },
 
-        cancel(event) {
-            event.target.disabled = true;
-            this.eventHub.$emit("close-task-modal");
+        cancel() {
+
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/AddTaskData.vue:397]: remove extra line

File: src/resources/js/components/global/taskComponents/PlaceTask.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
+<template>
+
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/PlaceTask.vue:2]: remove extra lines

File: src/resources/js/components/global/taskComponents/PlaceTask.vue [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        </div>
+        <button @click="assignTask($event)"
+                class="px-4 py-3 border border-transparent rounded text-white bg-indigo-600 hover:bg-indigo-500 transition duration-300 ease-in-out"
+                type="button">
+            <span>Place Task</span>
+        </button>
+    </div>
+</template>
+
+
```
Comment (NicoChampagne) [inline src/resources/js/components/global/taskComponents/PlaceTask.vue:74]: extraleindsdklfsd

**Review by NicoChampagne — APPROVED** [review]
Approved, overall I noticed a lot of variability inside your vue files is terms on spacing, and line additions. I'd recommend selecting a the styling you would like in phpstorm for vue files, and enable autoformatting when leaving files. Therefore overtime the variability will decrease.

File: src/resources/js/components/global/taskComponents/AddTaskData.vue (hunk 3 above)
Comment (SiamakSamie) [reply src/resources/js/components/global/taskComponents/AddTaskData.vue:398]: fuck, yeah i'll do that

**Pull request merged**